types_test.go

```diff
@@ -628,6 +628,84 @@ func TestIsHostname(t *testing.T) {
 	}
 }
 
+func TestIsGCPProjectId(t *testing.T) {
+	tests := []struct {
+		name   string
+		target string
+		want   bool
+	}{
+		{
+			name:   "GCP Project ID with length equal to or greater than 6 characters",
+			target: "rabbit",
+			want:   true,
+		},
+		{
+			name:   "GCP Project ID length should not less than 6 characters",
+			target: "hello",
+			want:   false,
+		},
+		{
+			name:   "GCP Project ID with length equal to or less than 30 characters",
+			target: "bagase-crucible-bubble-gorilla",
+			want:   true,
+		},
+		{
+			name:   "GCP Project ID length should be greater than 30 characters",
+			target: "thalamus-instinct-teaching-bemadden-word",
+			want:   false,
+		},
+		{
+			name:   "GCP Project ID should only contain lowercase ASCII letters",
+			target: "macabre-MONOXIDE-bluish-biped",
+			want:   false,
+		},
+		{
+			name:   "GCP Project ID should not start with numbers",
+			target: "007bond",
+			want:   false,
+		},
+		{
+			name:   "GCP Project ID should only end with letters or digits",
+			target: "inherent-derris-",
+			want:   false,
+		},
+		{
+			name:   "GCP Project ID should only hyphens in symbols",
```

Comment:
`s/GCP Project ID should only hyphens in symbols/GCP Project ID can only have hyphens and symbols`?